### PR TITLE
[Testing] Remove a few more custom configurations

### PR DESCRIPTION
**Pull request by nmittler**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by nmittler** [comment]
@howardjohn are we fine with enabling endpointslice by default now? If so, can we just get rid of the test?

**Comment by nmittler** [comment]
@howardjohn also, it appears that enabling endpoint slice breaks the security reachability tests: https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/27937/integ-ipv6-k8s-tests_istio/7608. Should we raise a bug for this?

File: tests/integration/pilot/analysis/main_test.go [hunk 1]
```diff
@@ -30,16 +29,6 @@ func TestMain(m *testing.M) {
 	framework.
 		NewSuite(m).
 		RequireSingleCluster().
-		Setup(istio.Setup(nil, func(_ resource.Context, cfg *istio.Config) {
```
Comment (howardjohn) [inline tests/integration/pilot/analysis/main_test.go:33]: can be followup - if we do this we should merge into the mainline pilot/ package

**Comment by nmittler** [comment]
@howardjohn raised https://github.com/istio/istio/issues/27946 to track the endpointslice issue ... assigned to you for now, feel free to reassign as appropriate.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**